setup.py

```diff
@@ -1,57 +1,35 @@
-import setuptools.command.build_py
-import distutils.command.build_py as orig
-import distutils.log
-import setuptools
-from subprocess import run as sub_run
-import shutil
+import logging
 import os
+import shutil
+from subprocess import run as sub_run
+from setuptools import setup
+from setuptools.command.build_py import build_py
 
 
-class BuildPyCommand(setuptools.command.build_py.build_py):
+class BuildPyCommand(build_py):
     """Custom build command to pyccelise _kernels files in the build directory."""
 
-    # Copy the setuptools.command.build_py.build_py.finalize_options to recreate the __updated_files variable
-    def finalize_options(self):
-        orig.build_py.finalize_options(self)
-        self.package_data = self.distribution.package_data
-        self.exclude_package_data = self.distribution.exclude_package_data or {}
-        if 'data_files' in self.__dict__:
-            del self.__dict__['data_files']
-        self.__updated_files = []
-
-    # Rewrite the build_module function to copy each module in the build repository and pyccelise the modules ending with _kernels
+    # Rewrite the build_module function to copy each module in the build
+    # repository and pyccelise the modules ending with _kernels
     def build_module(self, module, module_file, package):
-        # This part is copied from distutils.command.build_py.build_module
-        if isinstance(package, str):
-            package = package.split('.')
-        elif not isinstance(package, (list, tuple)):
-            raise TypeError(
-                "'package' must be a string (dot-separated), list, or tuple"
-            )
-        outfile = self.get_module_outfile(self.build_lib, package, module)
-        dirname = os.path.dirname(outfile)
-        self.mkpath(dirname)
-        outfile, copied = self.copy_file(module_file, outfile, preserve_mode=0)
-
+        outfile, copied = super().build_module(module, module_file, package)
 
-        # This part check if the module is pyccelisable and pyccelise it in case
+        # This part check if the module is pyccelisable and pyccelise it in
+        # case
         if module.endswith('_kernels'):
-            self.announce(
-                '\nPyccelise module: %s' % str(module),
-                level=distutils.log.INFO)
-            sub_run([shutil.which('pyccel'), outfile, '--language', 'fortran', '--openmp'], shell=False)
-
-        # This part is copy from setuptools.command.build_py.build_module
-        if copied:
-            self.__updated_files.append(outfile)
+            self.announce(f"\nPyccelise module: {module}", level=logging.INFO)
+            sub_run([shutil.which('pyccel'), outfile,
+                     '--language', 'fortran',
+                     '--openmp'],
+                    shell=False, check=True) # nosec B603
```

Comment:
This change reveals **a hidden bug in macOS CI test**. It fails `pyccel`-izing kernel files since `gfortran` is missing ([logfile](https://github.com/pyccel/psydac/actions/runs/9000566960/job/24724996688)):

```
Building wheels for collected packages: psydac, igakit
  Building wheel for psydac (pyproject.toml): started
  Building wheel for psydac (pyproject.toml): finished with status 'error'
  error: subprocess-exited-with-error
  
  × Building wheel for psydac (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [84 lines of output]
      running bdist_wheel
      running build
      8< --- LOG TRIMMED --- 8< 
      copying psydac/core/bsplines_kernels.py -> build/lib/psydac/core
      
      Pyccelise module: bsplines_kernels
      
      ERROR at Fortran compilation stage
      pyccel:
       |fatal [codegen]: bsplines_kernels.py| Could not find compiler (gfortran)
```

This is fixed by `brew reinstall gcc` (9015ccc5), which ensures that  `gfortran` is properly installed.

Reply:
Good catch!

Reply:
Do you know if it possible to add some print statements during the installation process, in order to show that the kernel files are pyccelized? Currently these are only shown with `pip install . --verbose`, which prints lots of additional information.

Reply:
`subprocess.sub_run()` can be replaced by `subprocess.Popen()` to capture output of `pyccel` in realtime. I can modify this on the next commit.

Reply:
> `subprocess.sub_run()` can be replaced by `subprocess.Popen()` to capture output of `pyccel` in realtime. I can modify this on the next commit.

Unfortunately this doesn't work and I'm not able to limit `pip` verbosity to `pyccel`.  The most likely reason is that every module logs at [logging.INFO](https://github.com/numpy/numpy/blob/67a04b100b9de75c72814701d0437f48f6651cb4/numpy/distutils/log.py#L67-L77) verbosity, and we're not able to silence those output.

Should we do  `pip install psydac --verbose` in CI? It seems be quite useful in detecting odd behavior that couldn't be tested easily. However combing through the CI logs would become more annoying 😅

Reply:
I think we can leave things as they are for now. If we run into an installation issue we can always use the `--verbose` flag to get more information.